pkgs/development/python-modules/Theano/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv
+, runCommandCC
+, fetchFromGitHub
+, buildPythonPackage
+, pythonOlder
+, numpy
+, six
+, scipy
+, nose
+, nose-parameterized
+, pydot_ng
+, sphinx
+, pygments
+, future
+, cudaSupport ? false
+, cudatoolkit ? null
+, pycuda ? null
+, libgpuarray ? null
+, cudnn ? null
+}:
+
+assert cudaSupport
+  -> cudatoolkit != null
+  && libgpuarray != null
+  && pycuda != null
+  && cudnn != null;
+
+buildPythonPackage rec {
+  name = "Theano-cuda-${version}";
+  version = "0.8.2";
+
+  src = fetchFromGitHub {
+    owner = "Theano";
+    repo = "Theano";
+    rev = "46fbfeb628220b5e42bf8277a5955c52d153e874";
+    sha256 = "1sl91gli3jaw5gpjqqab4fiq4x6282spqciaid1s65pjsf3k55sc";
+  };
+
+  doCheck = false;
+
+  patches = [ ./compiler.patch ];
+  postPatch =
+    let
+      # Closure for running a command in a stdenv for building with given buildInputs
+      wrapped = command: buildInputs:
+        runCommandCC "${command}-wrapped" { inherit buildInputs; } ''
+          cat > "$out" <<EOF
+          #!$(type -p bash)
+          $(declare -xp | sed -e '/^[^=]\+="\('"''${NIX_STORE//\//\\/}"'\|[^\/]\)/!d')
+          $(type -p ${command}) "\$@"
+          EOF
+          chmod +x "$out"
+        '';
+
+    in ''
+      sed -i -e 's|@g++@|${wrapped "g++" [ libgpuarray ]}|' theano/configdefaults.py
```

Comment:
Might be nicer to create a new variable, say `compiler = ${wrapped "g++" [ libgpuarray ]}` and then modify the patch to use `@compiler@` (or `@stdenv.cc@`).